versioned/spi/src/main/java/org/projectnessie/versioned/TracingVersionStore.java

```diff
@@ -253,27 +253,20 @@ static String makeSpanName(String name) {
       Consumer<SpanBuilder> spanBuilder,
       InvokerWithOneException<Stream<R>, E1> invoker)
       throws E1 {
-    SpanHolder span = createSpan(spanName, spanBuilder);
     // We keep the span active (in scope) only for the duration of the call that creates the stream,
     // but close it when the stream is closed. This is because the outer scopes may call other
     // instrumented code during stream iteration, but those calls should be nested directly under
     // the caller's span, not under this `span`. This may cause overlapping spans, but at the same
     // time it allows tracking the total duration of the stream iteration process.
-    Stream<R> result = null;
-    try (@SuppressWarnings("unused")
-        Scope autoclosed = activeScope(span.get())) {
-      result = invoker.handle();
-      return result.onClose(span::close);
-    } catch (IllegalArgumentException e) {
-      // IllegalArgumentException is a special kind of exception that indicates a user-error.
-      throw e;
-    } catch (RuntimeException e) {
-      throw traceError(span.get(), e);
-    } finally {
-      // We cannot `catch (E1 e)`, so assume that the delegate threw an exception, when result==null
-      // and then close the span.
-      if (result == null) {
-        span.close();
+    try (SpanHolder span = createSpan(spanName + ".stream", spanBuilder);
+        Scope ignore = activeScope(span.get())) {
+      try {
+        return invoker.handle();
```

Comment:
I'm ok with this change, but I think the comment above is no longer 100% accurate because now we close the span before returning from this method.

Reply:
For background: my idea for the old impl. was to keep the span open until the stream is closed to allow tracing visualizations to display the whole duration of the stream processing as an "async" operation. I would not expect nested traced calls to be contained within the stream span, but I might be been mistaken :)

Reply:
Well, there are usually nested calls that are also traced. It's usually not a problem when only the VersionStore's traced, but with this change, the probability is nearly 100% that consuming ("starting") the stream causes nested activity.